src/core/decoder/src/decode/huffman.c

```diff
@@ -21,15 +21,15 @@
 
 /*- General utility functions -------------------------------------------------------------------*/
 
-static inline uint8_t clz(int32_t streamData, uint8_t numBits)
+static inline uint8_t clz(uint32_t streamData, uint8_t numBits)
 {
 #if defined WIN32
     return (uint8_t)(_lzcnt_u32(streamData) + numBits - 32);
 #elif VN_COMPILER(GCC) && (__GNUC__ >= 14)
     /* Annoyingly, it's undefined behaviour if you provide 0 to __builtin_clz. To match the Windows
      * behaviour (where 0 is just another leading zero), we need clzg, with sizeof(streamData) as
      * the default arg. */
-    return (uint8_t)(__builtin_clzg(streamData, (sizeof(streamData) * 8)) + numBits - 32);
+    return (uint8_t)(__builtin_clzg(streamData, (int) (sizeof(streamData) * 8)) + numBits - 32);
```

Comment:
Maybe `int32_t` rather than `(int)`?

Also, looking at this, I wonder if it's better to provide `numBits` (or indeed `(int32_t) numBits`) directly, as the second argument. That would make it more obviously identical to the pre-gcc-14 code..

Reply:
From reading https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-_005f_005fbuiltin_005fclzg and https://clang.llvm.org/docs/LanguageExtensions.html#builtin-clzg-and-builtin-ctzg  it says the second argument is specifically an int.

Reply:
Fair, may as well keep it consistent.